src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -363,15 +361,15 @@ public void observe(@NonNull SCMHead head, @NonNull SCMRevision revision) {
     }
 
     private void scheduleBuild(BranchProjectFactory<P,R> factory, final P item, SCMRevision revision, TaskListener listener, String name) {
-        if (ParameterizedJobMixIn.scheduleBuild2(item, 0, new CauseAction(new SCMTrigger.SCMTriggerCause("Branch indexing"))) != null) {
+        if (ParameterizedJobMixIn.scheduleBuild2(item, 0, new CauseAction(new BranchIndexingCause())) != null) {
             listener.getLogger().println("Scheduled build for branch: " + name);
             try {
                 factory.setRevisionHash(item, revision);
             } catch (IOException e) {
                 e.printStackTrace(listener.error("Could not update last revision hash"));
             }
         } else {
-            listener.getLogger().println("Failed to schedule build for branch: " + name);
+            listener.getLogger().println("Did not schedule build for branch: " + name);
```

Comment:
In the context of a possible alternative to #1, this message is more accurate.

A null return value could mean several things (unfortunately `ScheduleResult.Refused` does not differentiate them); the most innocuous would be a `QueueDecisionHandler` that returns `false` from `shouldSchedule` (again, unfortunately, it is not given a `TaskListener` to explain its decision, or offered to return a `CauseOfBlockage` or similar).